language_server/src/lsp/feature_language_features.cpp

```diff
@@ -244,5 +247,86 @@ void feature_language_features::semantic_tokens(const json& id, const json& para
+{
+    File = 1,
+    Module = 2,
+    Namespace = 3,
+    Package = 4,
+    Class = 5,
+    Method = 6,
+    Property = 7,
+    Field = 8,
+    Constructor = 9,
+    Enum = 10,
+    Interface = 11,
+    Function = 12,
+    Variable = 13,
+    Constant = 14,
+    String = 15,
+    Number = 16,
+    Boolean = 17,
+    Array = 18,
+    Object = 19,
+    Key = 20,
+    Null = 21,
+    EnumMember = 22,
+    Struct = 23,
+    Event = 24,
+    Operator = 25,
+    TypeParameter = 26
+};
+
+
+const std::unordered_map<parser_library::document_symbol_kind, lsp_document_symbol_item_kind>
+    document_symbol_item_kind_mapping { { parser_library::document_symbol_kind::DAT,
+                                            lsp_document_symbol_item_kind::Array },
+        { parser_library::document_symbol_kind::EQU, lsp_document_symbol_item_kind::Boolean },
+        { parser_library::document_symbol_kind::MACH, lsp_document_symbol_item_kind::Constant },
+        { parser_library::document_symbol_kind::UNKNOWN, lsp_document_symbol_item_kind::Operator },
+        { parser_library::document_symbol_kind::VAR, lsp_document_symbol_item_kind::Variable },
+        { parser_library::document_symbol_kind::SEQ, lsp_document_symbol_item_kind::String },
+        { parser_library::document_symbol_kind::COMMON, lsp_document_symbol_item_kind::Struct },
+        { parser_library::document_symbol_kind::DUMMY, lsp_document_symbol_item_kind::Class },
+        { parser_library::document_symbol_kind::EXECUTABLE, lsp_document_symbol_item_kind::Object },
+        { parser_library::document_symbol_kind::READONLY, lsp_document_symbol_item_kind::Enum },
+        { parser_library::document_symbol_kind::MACRO, lsp_document_symbol_item_kind::File } };
+
+json feature_language_features::document_symbol_item_json(hlasm_plugin::parser_library::document_symbol_item symbol)
+{
+    return { { "name", symbol.name() },
+        { "kind", document_symbol_item_kind_mapping.at(symbol.kind()) },
+        { "range", range_to_json(symbol.symbol_range()) },
+        { "selectionRange", range_to_json(symbol.symbol_selection_range()) },
+        { "children", document_symbol_list_json(symbol.children()) } };
+}
+
+json feature_language_features::document_symbol_list_json(
+    hlasm_plugin::parser_library::document_symbol_list symbol_list)
+{
+    json result = json::array();
+    for (const auto& symbol : symbol_list)
+    {
+        result.push_back(document_symbol_item_json(symbol));
+    }
+    return result;
+}
+
+void feature_language_features::document_symbol(const json& id, const json& params)
+{
+    auto document_uri = params["textDocument"]["uri"].get<std::string>();
+
+    auto symbol_list = ws_mngr_.document_symbol(uri_to_path(document_uri).c_str());
+
+    response_->respond(id, "", document_symbol_list_json(symbol_list));
+
+    std::ofstream a;
+    a.open("/home/malimi/a.json", std::ofstream::trunc);
+    a << document_symbol_list_json(symbol_list);
+    a << "\n\n\n";
+    a.close();
```

Comment:
```suggestion
```

Reply:
...sorry. Forgot to delete that and a few other things.